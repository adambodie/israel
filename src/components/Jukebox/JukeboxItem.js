import React, { Component } from 'react';

export default class JukeboxItem extends Component {
	constructor(props) {
		super(props);
		this.state = {view: 'front'};
	}
	clickHandler(event) {
		if (this.state.view === 'front') {
			this.setState({view: 'back'});
		} else {
			this.setState({view: 'front'});
		}
	}
    render() {
		let className = 'flipper';
		let testName = 'front-view';
		if (this.state.view === 'back') {
			className = className + ' flipped';
			testName = 'back-view';

		}
		const { title, src } = this.props.data;
		return (
			<div className={className} data-testid={testName} >
				<div className='card'>
					<button onClick={this.clickHandler.bind(this)} className="jukebox-button">o</button>
					<div className='front'>
						<p>{title}</p>
					</div>
					<div className='back'>
						<iframe title={title} src={`https://www.youtube.com/embed/${src}`} frameBorder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowFullScreen></iframe>													
					</div>
				</div>
			</div>
		);
	}
};

