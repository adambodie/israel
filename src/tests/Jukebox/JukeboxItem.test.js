import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import JukeboxItem from '../../components/Jukebox/JukeboxItem'
import { StateMock } from '@react-mock/state'
import {
	render,
    cleanup,
    fireEvent,
    waitForElement
  } from 'react-testing-library'
  
  afterEach(cleanup)

it('renders without crashing', () => {
	const div = document.createElement('div')
	ReactDOM.render(<JukeboxItem data={{title: 'title', src: 'src'}}/>, div)
	ReactDOM.unmountComponentAtNode(div)
})

it('renders correctly', () => {
	const tree = renderer.create(<JukeboxItem data={{title: 'title', src: 'src'}}/>).toJSON()
	expect(tree).toMatchSnapshot()
})

it('clickHandler to have been called', () => {
	const onClick = jest.spyOn(JukeboxItem.prototype, 'clickHandler')
	const { getByText } = render(<JukeboxItem data={{title: 'title', src: 'src'}}/>)
	fireEvent.click(getByText('o'))
	expect(onClick).toHaveBeenCalled()
})


const renderComponent = ({ view }) =>
  render(
    <StateMock state={{ view }}>
      <JukeboxItem data={{title: 'title', src: 'src'}}/>
    </StateMock>
  )

  it('renders initial view', async () => {
    const { getByTestId } = renderComponent({ view: 'front' })
    await waitForElement(() => getByTestId(/front-view/i))
  })

  it('update view state', async () => {
    const { getByTestId, getByText } = renderComponent({ view: 'front' })
    fireEvent.click(getByText('o'))
    await waitForElement(() => getByTestId(/back-view/i))
  })