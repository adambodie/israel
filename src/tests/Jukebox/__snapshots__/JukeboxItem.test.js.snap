// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flipper"
  data-testid="front-view"
>
  <div
    className="card"
  >
    <button
      className="jukebox-button"
      onClick={[Function]}
    >
      o
    </button>
    <div
      className="front"
    >
      <p>
        title
      </p>
    </div>
    <div
      className="back"
    >
      <iframe
        allow="accelerometer autoplay encrypted-media gyroscope picture-in-picture"
        allowFullScreen={true}
        frameBorder="0"
        src="https://www.youtube.com/embed/src"
        title="title"
      />
    </div>
  </div>
</div>
`;
